src/utils.ts

```diff
@@ -160,8 +160,10 @@ export type AbiParameterToPrimitiveType<
   ? TAbiParameter['type'] extends infer TAbiType extends string
     ? Error<`Unknown type '${TAbiType}'.`>
     : never
-  : TAbiParameter['type'] extends `Error:${string}`
-  ? [TAbiParameter['type']]
+  : // 5. If we've gotten this far, let's check for errors in tuple components.
+  // (Happens for recursive tuple typed data types.)
+  TAbiParameter extends { components: Error<string> }
+  ? TAbiParameter['components']
```

Comment:
Felt better about overriding `components` than `type` since it's used more in the code. Curious what you think though!

Reply:
I think your approach feels better too. Also makes it feel more targeted to the specific thing that we are checking on.